src/wp-includes/blocks/index.php

```diff
@@ -12,6 +12,54 @@
 require BLOCKS_PATH . 'widget-group.php';
 require BLOCKS_PATH . 'require-dynamic-blocks.php';
 
+/**
+ * Registers core block types styles.
+ *
+ * @since 6.3.0
+ */
+function register_core_block_types_styles() {
+	static $core_blocks_meta;
+	if ( ! $core_blocks_meta ) {
+		$core_blocks_meta = require ABSPATH . WPINC . '/blocks/blocks-json.php';
+	}
+
+	$suffix = SCRIPT_DEBUG ? '' : '.min';
+
+	$style_fields = array(
+		'editorStyle',
+		'style',
+	);
+
+	foreach ( $core_blocks_meta as $name => $schema ) {
+		foreach ( $style_fields as $style_field ) {
+			if ( ! isset( $schema[ $style_field ] ) ) {
+				$style_handle = 'style' === $style_field ? "wp-block-{$name}" : "wp-block-{$name}-editor";
+				wp_register_style(
+					$style_handle,
+					false
+				);
+				continue;
+			}
```

Comment:
Why is this needed? In `register_block_style_handle()`, if the field doesn't exist in the block metadata, it just returns `false` and doesn't call `wp_register_style()`.

The logic used to potentially register with `false` URI is different in `register_block_style_handle()`, we should make sure to follow that here.

Reply:
This confused me as well, but consider. https://github.com/WordPress/wordpress-develop/commit/716a99587c04e5b37b4ceacc3ccce6c72c842464 / [https://core.trac.wordpress.org/ticket/53358 #53358] / https://github.com/WordPress/gutenberg/pull/32275/. 

Basically what happens in `register_block_style_handle` use realpath to check if the file exists. Look at this snippet 

https://github.com/WordPress/wordpress-develop/blob/fbef5030f67b6837119e9c890288bb0ace796940/src/wp-includes/blocks.php#L225-L255

If the file doesn't exist, then sets `$style_uri` to false. This is so that developers can add inline styles to these style sheets and style core blocks this way. Core itself also use these styles, see wp_common_block_scripts_and_styles. Example 

https://github.com/WordPress/wordpress-develop/blob/fbef5030f67b6837119e9c890288bb0ace796940/src/wp-includes/script-loader.php#L2371

With this, lots of themes / plugins and core itself would break. It is confusing, but it is needed. Maybe we should add a comment explaining it...

Reply:
@spacedmonkey I understand, however I am referring more specifically to the exact logic: There are three possible scenarios technically:
1. The `style`/`editorStyle` properties are defined in the schema _and_ the files exist.
2. The `style`/`editorStyle` properties are defined in the schema, but the files do not exist.
3. The `style`/`editorStyle` properties are not defined in the schema.

Right now, the PR registers the stylesheet with `false` in case 3 above. However, `register_block_style_handle()` does nothing in case 3, it bails early. Instead, it registers the stylesheet with `false` in case 2 above. So that is where your implementation and the current one in core differ.

I am not entirely sure how relevant that is for core blocks. Can you do some testing with your PR and check which core blocks (if any) the condition in https://github.com/WordPress/wordpress-develop/pull/4621/files#diff-c62beff71446448ff952e38805c8ae7652f4e2717d3ca41b1d9cd9963bd82464R44 applies to? That would help clarify whether this code is okay as is or whether it's safer to remove it.